src/consensus/pbft/pbft.h

```diff
@@ -446,6 +447,7 @@ namespace pbft
         global_commit_cb, &register_global_commit_ctx);
 
       auto rollback_cb = [](kv::Version version, RollbackInfo* rollback_info) {
+        rollback_info->store->get_history()->discard_pending();
```

Comment:
this is similar to the above, discard should be part of rollback, and only discarding the versions that are >= to the target version of the rollback call

Reply:
discard is called from `history::rollback`, but I think the correct thing to do is to discard any versions that are `>` than the target version of the rollback, instead of `>=`